partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/TypeLattice.kt

```diff
@@ -266,41 +266,57 @@ internal class TypeLattice private constructor(
                 SYMBOL to explicit(),
             )
             graph[DECIMAL] = relationships(
-                INT8 to explicit(),
-                INT16 to explicit(),
-                INT32 to explicit(),
-                INT64 to explicit(),
                 BOOL to explicit(),
-                DECIMAL to explicit(),
-                DECIMAL_ARBITRARY to explicit(),
+                INT8 to unsafe(),
+                INT16 to unsafe(),
+                INT32 to unsafe(),
+                INT64 to unsafe(),
+                INT to unsafe(),
+                DECIMAL to coercion(),
+                DECIMAL_ARBITRARY to coercion(),
                 FLOAT32 to explicit(),
                 FLOAT64 to explicit(),
                 STRING to explicit(),
                 SYMBOL to explicit(),
             )
-            graph[FLOAT32] = relationships(
+            graph[DECIMAL_ARBITRARY] = relationships(
                 BOOL to explicit(),
-                DECIMAL to explicit(),
+                INT8 to unsafe(),
+                INT16 to unsafe(),
+                INT32 to unsafe(),
+                INT64 to unsafe(),
+                INT to unsafe(),
+                DECIMAL to unsafe(),
```

Comment:
Let's make those explicit as well, for consistent with DECIMAL since right now we are kind of inflating those two types during inferring.